strawberry/subscriptions/protocols/graphql_transport_ws/handlers.py

```diff
@@ -117,6 +119,12 @@ async def handle_message(self, message: dict):
         await self.reap_completed_tasks()
 
     async def handle_connection_init(self, message: ConnectionInitMessage) -> None:
+        if message.payload is not UNSET and not isinstance(message.payload, dict):
```

Comment:
The `message.payload is not UNSET` case is currently only implicitly tested by the pre-existing tests. Personally, I would add explicit tests for that as well :)

Reply:
Good idea! I've added some tests that I believe handle this case, which are ready to review again now :D